.github/workflows/release.yml

```diff
@@ -0,0 +1,75 @@
+name: Release
+
+on:
+  push:
+    tags: [ '*' ]
+
+jobs:
+  test-and-release:
+    # Clone of main test workflow
+    runs-on: macos-10.15
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Ruby setup
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6
+
+    - name: Bundle cache
+      uses: actions/cache@v2
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-gems-
+    - name: Bundle install
+      run: |
+        gem install bundler
+        bundle config path vendor/bundle
+        bundle install --jobs 4 --retry 3
+
+    - name: Cocoapods cache
+      uses: actions/cache@v2
+      with:
+        path: Pods
+        key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-pods-
+    - name: Cocoapods install
+      run: |
+        bundle exec fastlane run cocoapods
+
+    - name: Java setup
+      uses: actions/setup-java@v1
+      with:
+        java-version: '11'
+    - name: WireMock setup
+      run: |
+        curl https://repo1.maven.org/maven2/com/github/tomakehurst/wiremock-jre8-standalone/2.27.0/wiremock-jre8-standalone-2.27.0.jar -o wiremock.jar -s
+        bundle exec iostrust add ./wiremock/cert/wiremock.crt
+        java -jar wiremock.jar --https-port 9099 --root-dir wiremock --https-keystore wiremock/cert/wiremock.jks --keystore-password password --verbose &
+
+    - name: Test
```

Comment:
Do you think it is necessary to test just before releasing ?
Normally, everything on the main branch should be releasable. And if there is a flaky tests that turn red, then the release will be abort and we would need to delete the remote tag and re-push it.

Please confirm if you're confident on tests, I will +2 then.

Reply:
Not all commits are merged / pushed on main and therefore you cannot guarantee all will succeed at tests. For instance when you rebase / merge, only the last commit is tested. In any cases, It can't hurt to test a last time before releasing.
I'm confident on tests, and anyway I think you can retriever workflow manually if needed.